### DISCUSSION TITLE
Unable to activate source error

### DISCUSSION THREAD

**Discussion by MalekChebli** [nfstream/nfstream#111] [Q&A] [answered]
Hello, I'm currently a student working on a DPI project and I'm using NFStream as a nDPI tool.
Whenever I put a network interface as a source it shows me this error. 
Is NFStream able to read directly from a network interface and carry on the process of parsing or just from a pcap file ? 
![image](https://user-images.githubusercontent.com/100672682/156325122-9afa24ef-2eb4-4cda-861b-9f653992fde0.png)

**Comment by drnpkr** [comment]
Hi @MalekChebli

If you want to access your interface from the OS, you have to use sudo. There is a workaround to access the packets without using sudo but it is not trivial. I suggest you run your code in sudo elevated mode. Please note that the Python installation for your sudo might be different than that installed under your pfedpdk user.

**Reply by MalekChebli** [reply]
thanks for your quick answer.

**Comment by MalekChebli** [comment, marked as answer]
Good morning,
I would like to ask you another question. Does NFStream support real-time traffic capture and packets parsing ?

**Reply by drnpkr** [reply]
Sure. It does. The term real-time is relative to the active and idle (passive) timeouts you set. The default idle timeout for NFStream is 120 seconds. This means that if there is no packets captured in a flow for 120 seconds, the flow is considered expired and sent for further processing. In contrast, active timeout expires the low after the specified duration regardless there are packets observed in the flow.

You might wonder why NFStream does nothing although it is started in live capture mode. It does, but you did not wait enough. Considering the above mentioned, only after 120 seconds the first flows are going to show up in your program. Exactly when the idle timeout has expired. Note that you might lower these timeouts, however, it can increase the burden on the hardware causing packet drops.
